util/dvsim/doc/testplanner.md

```diff
@@ -0,0 +1,285 @@
+---
+title: "testplanner tool"
+---
+
+testplanner is a Python based tool for parsing testplans written in Hjson
+format into a data structure that can be used for:
+* Expanding the testplan inline within the DV document as a table
+* Annotating the simulation results with testplan entries for a document driven
+  DV execution
+
+Please see [DV methodology]({{< relref "doc/ug/dv_methodology/index.md#documentation" >}})
+for more details on the rationale and motivation for writing and maintaining testplans
+in a machine-parsable format (`Hjson`).
+This document will focus on the anatomy of a Hjson testplan, list of features supported
+and some of the ways of using the tool.
+
+## Hjson testplan
+
+A testplan consists of a list of testpoints (planned tests) and a list of
+planned coverage items (covergroups).
+
+### Testpoints
+
+A testpoint is an entry in the testplan representing a planned test. A planned
+test maps 1:1 to a uniquely testable feature of the design. Minimally, the
+following items are sufficient to adequately capture the intent of a planned
+test:
+* **name: name of the planned test**
+
+    This is a single `lower_snake_case` string that succinctly describes the intended
+    feature being tested. As an example, a smoke test which is typically the
+    first test written on a brand new testbench would be simply named `smoke`.
+
+* **milestone: verification milestone**
+
+    This is one of {"`V1`", "`V2`" and "`V3`"}. This allows us to concretely indicate
+    that all goals for a particular milestone have been achieved and we can
+    transition to the next.
+
+* **desc: description of the planned test**
+
+    A multi-line string that briefly describes the intent of the test. It is
+    recommended to add a high level goal, stimulus and checking procedure so
+    that the reader gets the full picture of what and how the said feature is being
+    tested.
+
+    Full [Markdown]({{< relref "doc/rm/markdown_usage_style" >}}) syntax is supported
+    when writing the description.
+
+* **tests: list of actual written tests that maps to this planned test**
+
+    The testplan is written in the initial work stage of the verification
+    [life-cycle]({{< relref "doc/project/development_stages#hardware-verification-stages" >}}).
+    When the DV engineer gets to actually developing the test, it may not map 1:1 to
```

Comment:
should that be "When the DV engineer gets to actually develop the test"

Reply:
I think Sri's text is right. It's short for "gets around to actually developing" (whereas your version means "is allowed to actually develop")

Reply:
haha thanks Rupert for explaining it. I am fine either way, could be just my bad english haha.

Reply:
Grammar-wise this is incorrect: remove the adverb and it says "gets to developing the", which is certainly wrong. I am a big fan of E. B. White "The Elements of Style" (great read by the way), and it would recommend rewriting this as "... engineer develops the ...", so my other comment to this line is also undesirable.